test/e2e/singlecluster/statefulset_test.go

```diff
@@ -0,0 +1,85 @@
+package e2e
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/util/kubeversion"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	statefulsettesting "sigs.k8s.io/kueue/pkg/util/testingjobs/statefulset"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("Stateful set integration", func() {
+	var (
+		ns         *corev1.Namespace
+		onDemandRF *kueue.ResourceFlavor
+	)
+
+	ginkgo.BeforeEach(func() {
+		if kubeVersion().LessThan(kubeversion.KubeVersion1_27) {
```

Comment:
This should not be needed since we only run tests on 1.28+ now. Please also open a follow up PR to remove the checks from other places .